web/cobrands/fixmystreet/staff.js

```diff
@@ -123,24 +123,21 @@ fixmystreet.staff_set_up = {
                 $emailOptionalLabel.removeClass('hidden');
                 $emailInput.removeClass('required');
             }
-            $nameInput.val('').prop('disabled', false);
-            $nameInput.val('').prop('readonly', false);
+            $nameInput.val('').prop('disabled', false).prop('readonly', false);
             $phoneInput.val('').prop('disabled', false);
             $showNameCheckbox.prop('checked', false).prop('disabled', true);
             $addAlertCheckbox.prop('checked', true).prop('disabled', false);
         } else if (val === 'anonymous_user') {
             $emailInput.val('-').prop('disabled', true);
             $emailOptionalLabel.addClass('hidden');
-            $nameInput.val('-').prop('disabled', true);
-            $nameInput.val('').prop('readonly', false);
+            $nameInput.val('-').prop('disabled', true).prop('readonly', false);
             $phoneInput.val('-').prop('disabled', true);
             $showNameCheckbox.prop('checked', false).prop('disabled', true);
             $addAlertCheckbox.prop('checked', false).prop('disabled', true);
         } else if (val === 'body') {
             $emailInput.val('-').prop('disabled', true);
             $emailOptionalLabel.addClass('hidden');
-            $nameInput.val(txt).prop('disabled', true);
-            $nameInput.val('').prop('readonly', false);
+            $nameInput.val(txt).prop('disabled', false).prop('readonly', true);
```

Comment:
@dracos I have updated this line to use 'readonly' instead of 'disabled' as the new tests weren't passing otherwise. However, previously `$nameInput` would be set to 'disabled' in this case and was fine. Is this an appropriate / acceptable change?

Reply:
I can't replicate this, the new tests pass fine with `disabled` so going to leave this as is.